src/aeat/rest_framework/serializers.py

```diff
@@ -1,83 +1,86 @@
-    return ctrl.request(data)
 
+class DequeToDictMixin:
+    def to_internal_value(self, data):
+        return deque_to_dict(data)
 
-class AEATRequest(rf.Serializer):
-    service_name = None
-    include_test_mode = True
 
-    def save(self):
-        return make_aeat_request(self.service_name, self.data)
+class ENSSerializer(DequeToDictMixin, rf.Serializer):
+    is_error = False
 
-    def to_representation(self, instance):
-        instance = super().to_representation(instance)
+    mrn = rf.CharField(source='HEAHEA.DocNumHEA5')
 
-        if self.include_test_mode and settings.AEAT_TEST_MODE:
-            instance['TesIndMES18'] = '0'
 
-        return instance
+class EXSSerializer(DequeToDictMixin, rf.Serializer):
+    is_error = False
 
+    mrn = rf.CharField(source='HEAHEA.DocNumHEA5')
+    item_number_involved = rf.IntegerField(source='RISANA.IteNumInvRKA1')
+    customs_intervention_code = rf.CharField(source='RISANA.CusIntCodRKA1')
 
-class ENSQuerySerializer(AEATRequest):
-    service_name = 'ens_query'
-    include_test_mode = False
 
-    TraModAtBorHEA76 = fields.RequiredStr(help_text='Transport mode at border. EG 1')
-    ExpDatOfArr = fields.RequiredStr(help_text='Estimated date of arrival. EG 20110809')
-    ConRefNum = fields.RequiredStr(help_text='Transport identifier. EG 9294408')
+class UnknownResponseSerializer(rf.Serializer):
+    is_error = True
 
+    def to_representation(self, obj):
+        return {'reason': 'Unknown AEAT response'}
 
-class ENSForkSerializer(AEATRequest):
-    service_name = 'ens_fork'
 
+class ENSFunctionalErrorSerializer(DequeToDictMixin, rf.Serializer):
+    is_error = True
 
-class ENSPresentationSerializer(v4.BaseV4Mixin, AEATRequest):
-    service_name = 'ens_presentation'
+    type = rf.CharField(source='FUNERRER1.ErrTypER11')
+    pointer = rf.CharField(source='FUNERRER1.ErrPoiER12')
+    reason = rf.CharField(source='FUNERRER1.OriAttValER14')
 
-    MesTypMES20 = fields.NotRequiredStr(default='CC315A', read_only=True,
-                                        help_text='Message type. EG CC315A')
-    HEAHEA = v4.ENSPresentationHeader(required=True)
 
+def parse_xsd(data):
+    # Try V2 Style
+    try:
+        xsd = data[0].nsmap[None]
+    except (IndexError, KeyError):
+        pass
+    else:
+        return xsd
 
-class ENSModificationSerializer(v4.BaseV4Mixin, AEATRequest):
-    service_name = 'ens_modification'
-    NOTPAR670 = v4.NotifyParty(required=True)
-    MesTypMES20 = fields.NotRequiredStr(default='CC313A', read_only=True,
-                                        help_text='Message type. EG CC313A')
-    HEAHEA = v4.ENSModificationHeader(required=True)
+    # Try V4 Style
+    try:
+        xsd = data[0].nsmap['ie']
+    except (IndexError, KeyError):
+        pass
+    else:
+        return xsd
 
 
-class EXSPresentationSerializer(v2.BaseV2Mixin, AEATRequest):
-    service_name = 'exs_common'
+def get_class_for_aeat_response(data):
```

Comment:
All unmapped responses will return an UnknownResponseSerializer with payload:

`{'reason': 'Unknown AEAT response'}`

Added ENSFunctionalErrorSerializer as example. It should be trivial to add more error serializers (2 missing for ENS and another 2/3 for EXS)